javascriptv3/example_code/s3/actions/get-object-lock-configuration.js

```diff
@@ -1,31 +1,58 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import { fileURLToPath } from "url";
 import {
   GetObjectLockConfigurationCommand,
   S3Client,
+  S3ServiceException,
 } from "@aws-sdk/client-s3";
 
 /**
- * @param {S3Client} client
- * @param {string} bucketName
+ * Gets the Object Lock configuration for a bucket.
+ * @param {{ bucketName: string }}
  */
-export const main = async (client, bucketName) => {
-  const command = new GetObjectLockConfigurationCommand({
-    Bucket: bucketName,
-    // Optionally, you can provide additional parameters
-    // ExpectedBucketOwner: "ACCOUNT_ID",
-  });
+export const main = async ({ bucketName }) => {
+  const client = new S3Client({});
 
   try {
-    const { ObjectLockConfiguration } = await client.send(command);
-    console.log(`Object Lock Configuration: ${ObjectLockConfiguration}`);
-  } catch (err) {
-    console.error(err);
+    const { ObjectLockConfiguration } = await client.send(
+      new GetObjectLockConfigurationCommand({
+        Bucket: bucketName,
+        // Optionally, you can provide additional parameters
+        // ExpectedBucketOwner: "<account ID that is expected to own the bucket>",
+      }),
+    );
+    console.log(
+      `Object Lock Configuration:\n${JSON.stringify(ObjectLockConfiguration)}`,
+    );
+  } catch (caught) {
+    if (
+      caught instanceof S3ServiceException &&
+      caught.name === "NoSuchBucket"
+    ) {
+      console.error(
+        `Error from S3 while getting object lock configuration for ${bucketName}. The bucket doesn't exist.`,
+      );
+    } else if (caught instanceof S3ServiceException) {
+      console.error(
+        `Error from S3 while getting object lock configuration for ${bucketName}.  ${caught.name}: ${caught.message}`,
+      );
+    } else {
+      throw caught;
+    }
```

Comment:
As mentioned above, this isn't a deal breaker but it's better to do:
```suggestion
  } catch (caught) {
    console.error(
        `Error from S3 while getting object lock configuration for ${bucketName}.`,
      );
    if (
      caught instanceof S3ServiceException &&
      caught.name === "NoSuchBucket"
    ) {
      console.error(
        `The bucket doesn't exist.`,
      );
    } else if (caught instanceof S3ServiceException) {
      console.error(
        `${caught.name}: ${caught.message}`,
      );
    } else {
      throw caught;
    }
```

Reply:
Oh I see what you're doing now. Eh... I don't hate it, but I don't love it. I don't typically see JS errors in a multi line format.